Source/WebCore/dom/Document.cpp

```diff
@@ -2282,6 +2282,31 @@ bool Document::updateStyleIfNeeded()
     return true;
 }
 
+ContentVisibilityForceLayoutScope::ContentVisibilityForceLayoutScope(const Element& context)
+{
+    if (context.document().view()) {
+        context.document().updateRelevancyOfContentVisibilityElements();
+        context.document().updateStyleIfNeeded();
+        if (context.renderer() && context.renderer()->style().skippedContentReason().has_value()) {
+            context.renderer()->setNeedsLayout();
+            m_document = context.document();
+            m_document->view()->layoutContext().forceSkippedContentLayout(true);
```

Comment:
The name `forceSkippedContentLayout` is slightly confusing, we're not forcing a layout, but setting a flag.

Maybe `setNeedsSkippedContentLayout()` && `needsSkippedContentLayout()`

Reply:
Yes, sounds more accurate, will have a stab at that tomorrow.